plugins/inputs/mavlink/mavlink.go

```diff
@@ -0,0 +1,126 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package mavlink
+
+import (
+	_ "embed"
+	"time"
+
+	"github.com/chrisdalke/gomavlib/v3"
+	"github.com/chrisdalke/gomavlib/v3/pkg/dialects/ardupilotmega"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// Plugin state
+type Mavlink struct {
+	// Config param
+	FcuURL                 string   `toml:"fcu_url"`
+	SystemID               uint8    `toml:"system_id"`
+	MessageFilter          []string `toml:"message_filter"`
+	StreamRequestEnable    bool     `toml:"stream_request_enable"`
+	StreamRequestFrequency int      `toml:"stream_request_frequency"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	// Internal state
+	connection *gomavlib.Node
+	acc        telegraf.Accumulator
+	loading    bool
+	terminated bool
+}
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (*Mavlink) SampleConfig() string {
+	return sampleConfig
+}
+
+// Container for a parsed Mavlink frame
+type metricFrameData struct {
+	name   string
+	tags   map[string]string
+	fields map[string]any
+}
+
+func (s *Mavlink) Start(acc telegraf.Accumulator) error {
+	s.acc = acc
+
+	// Start routine to connect to Mavlink and stream out data async
+	go func() {
+		endpointConfig, err := ParseMavlinkEndpointConfig(s)
+		if err != nil {
+			s.Log.Debugf("%s", err.Error())
+			return
+		}
```

Comment:
Config validation should be done in `func (s *Mavlink) Init() error` method.

Reply:
Fixed, move this code into Init.